pkg/controller/application/application_controller.go

```diff
@@ -133,9 +138,11 @@ func (r *ReconcileApplication) Reconcile(request reconcile.Request) (reconcile.R
 	return reconcile.Result{}, nil
 }
 
-func isApp(o metav1.Object) bool {
-	if o.GetLabels() == nil || !util.IsApplicationComponent(o.GetLabels()) {
-		return false
+func isApp(obs ...metav1.Object) bool {
+	for _, o := range obs {
+		if o.GetLabels() != nil && util.IsApplicationComponent(o.GetLabels()) {
```

Comment:
So, if any one of obs has required labels, return true, not all of them?

Reply:
Yes.